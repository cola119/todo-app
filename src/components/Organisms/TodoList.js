import React from 'react';

import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import HighIcon from '@material-ui/icons/NewReleases';
import MediumIcon from '@material-ui/icons/CheckBoxOutlineBlank';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import CheckIcon from '@material-ui/icons/Check';

import ListRow from '../Molecules/ListRow';

const getDate = (timestamp) => {
    const _d = new Date(timestamp);
    const Y = _d.getFullYear();
    const m = ("0" + (_d.getMonth() + 1)).slice(-2);
    const d = ("0" + _d.getDate()).slice(-2);
    const H = ("0" + _d.getHours()).slice(-2);
    const i = ("0" + _d.getMinutes()).slice(-2);
    const s = ("0" + _d.getSeconds()).slice(-2);
    return `${Y}/${m}/${d} ${H}:${i}:${s}`;
}

const TodoList = (props) => {
    const isCompleted = todo => todo.state === 'complete';
    const leftIcon = todo => isCompleted(todo) ? <CheckIcon /> : todo.priority === 'high' ? <HighIcon color="secondary" /> : <MediumIcon />;
    const doneStyle = {
        color: "gray",
        textDecoration: "line-through"
    }

    return (
        <List>
            <Divider />
            {props.todos.map(todo => (
                <ListRow
                    key={todo.id}
                    onRowClick={() => props.changeTodoState(todo.id)}
                    leftIcon={leftIcon(todo)}
                    primaryText={<span style={isCompleted(todo) ? doneStyle : {}}>{todo.title}</span>}
                    secondaryText={getDate(todo.created_at)}
                    secondaryAction={isCompleted(todo) &&
                        <IconButton onClick={() => props.deleteTodo(todo.id)}>
                            <DeleteIcon />
                        </IconButton>
                    }
                />
            ))}
        </List>
    );
};

export default TodoList;