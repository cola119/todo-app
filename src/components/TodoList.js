import React from 'react';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
// import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
// import Button from '@material-ui/core/Button';
import HighIcon from '@material-ui/icons/NewReleases';
import MediumIcon from '@material-ui/icons/CheckBoxOutlineBlank';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import CheckIcon from '@material-ui/icons/Check';

const get_date = (timestamp) => {
    const _d = new Date(timestamp);
    const Y = _d.getFullYear();
    const m = ("0" + (_d.getMonth() + 1)).slice(-2);
    const d = ("0" + _d.getDate()).slice(-2);
    const H = ("0" + _d.getHours()).slice(-2);
    const i = ("0" + _d.getMinutes()).slice(-2);
    const s = ("0" + _d.getSeconds()).slice(-2);
    return `${Y}/${m}/${d} ${H}:${i}:${s}`;
}

const TodoList = (props) => {
    const isCompleted = todo => todo.state === 'complete';

    const doneStyle = {
        color: "gray",
        textDecoration: "line-through"
    }

    return (
        <List component="nav">
            <Divider />
            {props.todos.map(todo => (
                <ListItem
                    key={todo.id}
                    divider button
                    onClick={() => props.changeTodoState(todo.id)}
                >
                    <ListItemIcon>
                        {isCompleted(todo) ? <CheckIcon /> : todo.priority === 'high' ? <HighIcon color="secondary" /> : <MediumIcon />}
                    </ListItemIcon>
                    <ListItemText
                        inset={true}
                        primary={
                            <span style={isCompleted(todo) ? doneStyle : {}}>{todo.title}</span>
                        }
                        secondary={
                            <>
                                {get_date(todo.created_at)}<br />
                                {/* {todo.deadline} */}
                            </>
                        }
                    />
                    <ListItemSecondaryAction>
                        {isCompleted(todo) &&
                            <IconButton onClick={() => props.deleteTodo(todo.id)}>
                                <DeleteIcon />
                            </IconButton>
                        }
                    </ListItemSecondaryAction>
                </ListItem>
            ))}
        </List>
    );
};

export default TodoList;