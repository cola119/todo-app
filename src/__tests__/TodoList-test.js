import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '../components/TodoList';

describe('<TodoList />', () => {
    it('<TodoList />のスナップショット', () => {
        const todos = [
            { id: 12345, title: "title1", priority: "high", deadline: "today", state: 'incomplete', created_at: 1519211810362 },
            { id: 23456, title: "title2", priority: "medium", deadline: "tomorrow", state: 'complete', created_at: 1519211811670 },
            { id: 65432, title: "title3", priority: "low", deadline: "week", state: 'incomplete', created_at: 1519211809934 },
            { id: 83553, title: "title4", priority: "medium", deadline: "other", state: 'complete', created_at: 1519129853500 },
            { id: 53215, title: "title5", priority: "high", deadline: "week", state: 'incomplete', created_at: 1519129864400 },
        ];

        const tree = renderer.create(<TodoList todos={todos} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})