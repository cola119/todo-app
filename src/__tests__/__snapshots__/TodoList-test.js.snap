// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> <TodoList />のスナップショット 1`] = `
<nav
  className="MuiList-root-1 MuiList-padding-2"
>
  <hr
    className="MuiDivider-root-5"
  />
  <li
    className="MuiListItem-container-11"
  >
    <div
      className="MuiButtonBase-root-22 MuiListItem-root-10 MuiListItem-default-13 MuiListItem-gutters-18 MuiListItem-divider-17 MuiListItem-button-19 MuiListItem-secondaryAction-20"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiListItemIcon-root-25"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-26 MuiSvgIcon-colorSecondary-28"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M23 12l-2.44-2.78.34-3.68-3.61-.82-1.89-3.18L12 3 8.6 1.54 6.71 4.72l-3.61.81.34 3.68L1 12l2.44 2.78-.34 3.69 3.61.82 1.89 3.18L12 21l3.4 1.46 1.89-3.18 3.61-.82-.34-3.68L23 12zm-10 5h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root-35 MuiListItemText-inset-36"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-subheading-48 MuiListItemText-primary-38"
        >
          <span
            style={Object {}}
          >
            title1
          </span>
        </span>
        <p
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-74 MuiListItemText-secondary-39"
        >
          2018/02/21 20:16:50
          <br />
        </p>
      </div>
      <span
        className="MuiTouchRipple-root-84"
      />
    </div>
    <div
      className="MuiListItemSecondaryAction-root-77"
    />
  </li>
  <li
    className="MuiListItem-container-11"
  >
    <div
      className="MuiButtonBase-root-22 MuiListItem-root-10 MuiListItem-default-13 MuiListItem-gutters-18 MuiListItem-divider-17 MuiListItem-button-19 MuiListItem-secondaryAction-20"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiListItemIcon-root-25"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-26"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root-35 MuiListItemText-inset-36"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-subheading-48 MuiListItemText-primary-38"
        >
          <span
            style={
              Object {
                "color": "gray",
                "textDecoration": "line-through",
              }
            }
          >
            title2
          </span>
        </span>
        <p
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-74 MuiListItemText-secondary-39"
        >
          2018/02/21 20:16:51
          <br />
        </p>
      </div>
      <span
        className="MuiTouchRipple-root-84"
      />
    </div>
    <div
      className="MuiListItemSecondaryAction-root-77"
    >
      <button
        className="MuiButtonBase-root-22 MuiIconButton-root-78"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-83"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-26"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-84"
        />
      </button>
    </div>
  </li>
  <li
    className="MuiListItem-container-11"
  >
    <div
      className="MuiButtonBase-root-22 MuiListItem-root-10 MuiListItem-default-13 MuiListItem-gutters-18 MuiListItem-divider-17 MuiListItem-button-19 MuiListItem-secondaryAction-20"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiListItemIcon-root-25"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-26"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root-35 MuiListItemText-inset-36"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-subheading-48 MuiListItemText-primary-38"
        >
          <span
            style={Object {}}
          >
            title3
          </span>
        </span>
        <p
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-74 MuiListItemText-secondary-39"
        >
          2018/02/21 20:16:49
          <br />
        </p>
      </div>
      <span
        className="MuiTouchRipple-root-84"
      />
    </div>
    <div
      className="MuiListItemSecondaryAction-root-77"
    />
  </li>
  <li
    className="MuiListItem-container-11"
  >
    <div
      className="MuiButtonBase-root-22 MuiListItem-root-10 MuiListItem-default-13 MuiListItem-gutters-18 MuiListItem-divider-17 MuiListItem-button-19 MuiListItem-secondaryAction-20"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiListItemIcon-root-25"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-26"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root-35 MuiListItemText-inset-36"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-subheading-48 MuiListItemText-primary-38"
        >
          <span
            style={
              Object {
                "color": "gray",
                "textDecoration": "line-through",
              }
            }
          >
            title4
          </span>
        </span>
        <p
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-74 MuiListItemText-secondary-39"
        >
          2018/02/20 21:30:53
          <br />
        </p>
      </div>
      <span
        className="MuiTouchRipple-root-84"
      />
    </div>
    <div
      className="MuiListItemSecondaryAction-root-77"
    >
      <button
        className="MuiButtonBase-root-22 MuiIconButton-root-78"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-83"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-26"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-84"
        />
      </button>
    </div>
  </li>
  <li
    className="MuiListItem-container-11"
  >
    <div
      className="MuiButtonBase-root-22 MuiListItem-root-10 MuiListItem-default-13 MuiListItem-gutters-18 MuiListItem-divider-17 MuiListItem-button-19 MuiListItem-secondaryAction-20"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiListItemIcon-root-25"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-26 MuiSvgIcon-colorSecondary-28"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M23 12l-2.44-2.78.34-3.68-3.61-.82-1.89-3.18L12 3 8.6 1.54 6.71 4.72l-3.61.81.34 3.68L1 12l2.44 2.78-.34 3.69 3.61.82 1.89 3.18L12 21l3.4 1.46 1.89-3.18 3.61-.82-.34-3.68L23 12zm-10 5h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root-35 MuiListItemText-inset-36"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-subheading-48 MuiListItemText-primary-38"
        >
          <span
            style={Object {}}
          >
            title5
          </span>
        </span>
        <p
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-74 MuiListItemText-secondary-39"
        >
          2018/02/20 21:31:04
          <br />
        </p>
      </div>
      <span
        className="MuiTouchRipple-root-84"
      />
    </div>
    <div
      className="MuiListItemSecondaryAction-root-77"
    />
  </li>
</nav>
`;
